dvc/render/convert.py

```diff
@@ -42,8 +44,10 @@ def _group_by_rev(datapoints):
 
 
 def to_json(renderer, split: bool = False) -> List[Dict]:
+    from copy import deepcopy
+
     if renderer.TYPE == "vega":
-        grouped = _group_by_rev(renderer.datapoints)
+        grouped = _group_by_rev(deepcopy(renderer.datapoints))
```

Comment:
Before the change we would edit the datapoints in renderer. That would result in vega json datapoints lacking `rev` field. We did not detect that before, because vscode nowadays uses only `--split` option and handles the datapoints by themselves.